src/UCTSearch.cpp

```diff
@@ -401,48 +402,50 @@ bool UCTSearch::is_running() const {
     return m_run && m_nodes < MAX_TREE_SIZE;
 }
 
-bool UCTSearch::stop_thinking(int elapsed_centis, int time_for_move) const {
-    auto visits = m_root->get_visits();
+int UCTSearch::est_playouts_left(int elapsed_centis, int time_for_move) const
+{
     auto playouts = static_cast<int>(m_playouts);
-    auto stop = playouts >= m_maxplayouts
-                || visits >= m_maxvisits
-                || elapsed_centis >= time_for_move;
-
-    if (stop || cfg_timemanage == TimeManagement::OFF) {
-        return stop;
+    const auto playouts_left = std::min(m_maxplayouts - playouts,
+                                        m_maxvisits - m_root->get_visits());
+    if (elapsed_centis > 100 && playouts > 100) {
+        // Wait for at least 1 second and 100 playouts
+        // so we get a reliable playout_rate.
+        const auto playout_rate = 1.0f * playouts / elapsed_centis;
+        const auto time_left = time_for_move - elapsed_centis;
+        return std::min(static_cast<float>(playouts_left), std::ceil(playout_rate * time_left));
+    }
+    else {
+        return playouts_left;
     }
+}
 
+int UCTSearch::prune_noncontenders(int elapsed_centis, int time_for_move) {
```

Comment:
Might be easier to have this return if all nodes have been trimmed

Reply:
1. From its name, the most straightforward expectation of what this function might return is the correct one. Anything different would be unintuitive and would at least have to be documented.
1. With this return value the function supplies more info, which might prove useful down the line.